src/audio/codec_adapter/codec_adapter.c

```diff
@@ -173,8 +173,24 @@ int codec_adapter_prepare(struct comp_dev *dev)
 	if (md->mpd.in_buff_size != mod->period_bytes)
 		mod->deep_buff_bytes = MIN(mod->period_bytes, md->mpd.in_buff_size) * buff_periods;
 
-	/* Allocate local buffer */
+	if (md->mpd.out_buff_size > mod->period_bytes) {
+		buff_periods = (md->mpd.out_buff_size % mod->period_bytes) ?
+			       (md->mpd.out_buff_size / mod->period_bytes) + 2 :
+			       (md->mpd.out_buff_size / mod->period_bytes) + 1;
+	} else {
+		buff_periods = (mod->period_bytes % md->mpd.out_buff_size) ?
+			       (mod->period_bytes / md->mpd.out_buff_size) + 2 :
+			       (mod->period_bytes / md->mpd.out_buff_size) + 1;
```

Comment:
Some readability notes. This same logic is repeating four time in this function, so a macro or inlined helper function would be helpful. Also reusing "buff_periods" for different purposes is a bit distracting. 

But alas, neither is added with this patch, so not really a blocker.

Reply:
Lets cleanup the readability in a subsequent update, I know @ranj063 has more in progress